server/cluster/cluster.go

```diff
@@ -1690,30 +1674,79 @@ func (c *RaftCluster) SetMinResolvedTS(storeID, minResolvedTS uint64) error {
 	return c.putStoreLocked(newStore)
 }
 
-func (c *RaftCluster) runMinResolvedTSJob(saveInterval time.Duration) {
-	defer c.wg.Done()
-	if saveInterval == 0 {
+func (c *RaftCluster) checkAndUpdateMinResolvedTS() {
+	c.Lock()
+	defer c.Unlock()
+
+	if !c.isInitialized() {
+		return
+	}
+	curMinResolvedTS := uint64(math.MaxUint64)
+	for _, s := range c.GetStores() {
+		if !core.IsAvailableForMinResolvedTS(s) {
+			continue
+		}
+		if curMinResolvedTS > s.GetMinResolvedTS() {
+			curMinResolvedTS = s.GetMinResolvedTS()
+		}
+	}
+	if curMinResolvedTS == math.MaxUint64 || curMinResolvedTS <= c.minResolvedTS {
 		return
 	}
+	c.minResolvedTS = curMinResolvedTS
+	c.storage.SaveMinResolvedTS(curMinResolvedTS)
+}
+
+func (c *RaftCluster) runMinResolvedTSJob() {
 	defer logutil.LogPanic()
-	ticker := time.NewTicker(saveInterval)
+	defer c.wg.Done()
+
+	interval := c.opt.GetMinResolvedTSPersistenceInterval()
+	if interval == 0 {
+		interval = DefaultMinResolvedTSPersistenceInterval
+	}
+	ticker := time.NewTicker(interval)
 	defer ticker.Stop()
+
+	c.loadMinResolvedTS()
 	for {
 		select {
 		case <-c.ctx.Done():
 			log.Info("min resolved ts background jobs has been stopped")
 			return
 		case <-ticker.C:
-			minResolvedTS := c.GetMinResolvedTS()
-			if minResolvedTS != math.MaxUint64 {
-				c.Lock()
-				c.storage.SaveMinResolvedTS(minResolvedTS)
-				c.Unlock()
+			interval = c.opt.GetMinResolvedTSPersistenceInterval()
+			if interval != 0 {
+				c.checkAndUpdateMinResolvedTS()
+			} else {
+				interval = DefaultMinResolvedTSPersistenceInterval
 			}
+			ticker.Reset(interval)
 		}
 	}
 }
 
+func (c *RaftCluster) loadMinResolvedTS() {
+	c.Lock()
```

Comment:
Ditto, I think we needn't holding the lock during `c.storage.LoadMinResolvedTS()` too cc @disksing @lhy1024